src/tools/tracejob.c

```diff
@@ -403,15 +405,21 @@ parse_log(FILE *fp, char *job, int ind)
 	strcpy(job_buf, job);
 
 	while (fgets(buf, buf_size, fp) != NULL) {
-		while(buf_size == strlen(buf) + 1){
-			buf = (char*)realloc(buf, buf_size * sizeof(char*));
-			tbuf = (char*)realloc(tbuf, buf_size * sizeof(char*));
-			if(buf == NULL || tbuf == NULL || fgets(tbuf, buf_size+1, fp) == NULL){
+		while(buf_size == strlen(buf) + incr){
```

Comment:
I added that to account for the fact that only n-1 characters are being read each time we call fgets(). So each iteration there is one less character ending up in buf; so I factored in that difference in length with incr.

I have changed that though; passing in tbuf_size + 1 to fgets().